docs/sources/write/style-guide/write-for-developers/index.md

```diff
@@ -78,23 +78,110 @@ Additional tips:
 Readers of documentation typically skim through it to find code samples they can copy and paste and run as is.
 Because of this and whenever possible, provide production-ready examples.
 
-However, it isn't necessary for every code example to be runnable in production. Some code examples are written to illustrate a point so that the developer can learn how to do something similar on their own.
+However, it isn't necessary for every code example to be runnable in production.
+Some code examples are written to illustrate a point so that the developer can learn how to do something similar on their own.
+Such examples should be clearly marked as partial.
 
 When providing an example, give a written description. You can put it either in the body of the document or as explanatory comments within the example code.
 
 Remember the basic rule: explain _why_ your code does what it does, rather than describe _what_ it does.
 For an in-depth, external resource about writing developer documentation, see [_Docs for Developers_](https://docsfordevelopers.com/).
 
-### Formatting code examples
+### Format code examples
 
-Here are guidelines to follow when formatting code examples.
+To format code examples, use the following guidelines.
 
-- Use spaces, not tabs.
-- Follow Grafana's accepted [coding style guidelines](https://github.com/grafana/grafana/blob/main/contribute/style-guides/).
-- Wrap lines at 80 characters.
-- When omitting code, use three dots (...). Don't use the ellipsis character (…).
+#### Configure syntax highlighting
 
-Introduce each code sample with a sentence or paragraph to establish its context. End the introduction with a colon if it immediately precedes the sample, or a period if it doesn't.
+In Markdown, an info string after the first three backticks is used to describe the language contained within.
+The website uses this information for syntax highlighting.
+
+To configure syntax highlighting for a JSON code example, refer to the following Markdown example:
```

Comment:
This isn't really an example of how to **configure** anything, because in practice you don't need to wrap your code examples in markdown for them to work.  I wrote a couple of different revisions for this line, still not happy with this one...

```suggestion
The following is an example of how syntax highlighting is applied to a JSON code example:
```